### PR TITLE
core/settings: add README

### PR DESCRIPTION
**Pull request by janiversen**
Add simple explanation of how the program flow is.

Signed-off-by: Jan Iversen <jani@apache.org>

### Describe the pull request:

- [x ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

add README in core/settings to describe workflow.

based on request from @dirkhh

### CODE REVIEW

**Comment by henrik242** [comment]
Cool! Would you be able to convert it to markdown?

File: core/settings/README [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+core/qt-init.cpp init_qt_late()
+	call qPref::load() to load all properties
+
+
+>> System shutdown:
+
+subsurface-mobile-main.cpp and subsurface-desktop-main.cpp main()
+	call qPref::sync() to save all changes to disk, this is needed
+	because part of the program modifies struct preferences insteaad
```
Comment (henrik242) [inline core/settings/README]: `insteaad` :)

**Comment by dirkhh** [comment]
That's great documentation to help people understand how things work.
I agree with both of @henrik242 's comments. Markdown makes it easier for people browsing the source on the web and typically is unobtrusive enough to not affect readability on the terminal.

**Comment by janiversen** [comment]
Ihave not done markdown for years, but I guess it is like riding a bicycle, let me have a look.

@dirkhh I actually had 2 commits here, one for the typo in qPrefUpdateManager but that went way when I rebased to the project master, seems you already made that change.

**Comment by dirkhh** [comment]
I fixed the typo - again, that would have potentially created a user visible issue. I didn't know how quickly you'd find time to look at it (based on your comments and emails to me), so I simply did it myself yesterday (it wasn't that hard, even for me). 
Here it is: 4bb72160a6323e1ab3f3d9ea2c7f90c21088347f

**Comment by janiversen** [comment]
I have converted the text to md blindfolded, so comments etc on how to make it more readable is welcome.

I also cherry-picked an update to qPrefDisplay() that solves the load issue (I am thinking of how to make a testcase for this, but it seems to requiere I declare the test case a friend).

**Comment by janiversen** [comment]
I have 0% problem with the fact that you fixed it, it was merely my surprise that git removed a commit without  even telling me.

**Comment by dirkhh** [comment]
> it was merely my surprise that git removed a commit without  even telling me.

I was actually surprised by that as well... Clearly the code of your commit was the same, but it would have had a different SHA... So I wouldn't have expected GitHub to just silently drop it...

-- 
From my phone

**Comment by dirkhh** [comment]
> Do not bother...I looked over the code, and see a couple of places where I can make it more efficient.

I took this to mean "please don't merge this PR" - if I misunderstood, please let me know and we can get this one merged.

**Comment by janiversen** [comment]
Ups that could indeed be misunderstood, I meant please fo not bother to revert the revert ! this PR is ready to be merged.

**Pull request merged**